pkg/disttask/framework/proto/step.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proto
+
+import "fmt"
+
+// Step is the step of task.
+type Step int64
+
+// TaskStep is the step of task.
+// DO NOT change the value of the constants, will break backward compatibility.
+// successfully task MUST go from StepInit to business steps, then StepDone.
+const (
+	StepInit Step = -1
+	StepDone Step = -2
+)
+
+// Step2Str converts step to string.
+// it's too bad that we define step as int 🙃.
+func Step2Str(t TaskType, s Step) string {
+	// StepInit and StepDone are special steps, we don't check task type for them.
+	if s == StepInit {
+		return "init"
+	} else if s == StepDone {
+		return "done"
+	}
+	switch t {
+	case Backfill:
+		return backfillStep2Str(s)
+	case ImportInto:
+		return importIntoStep2Str(s)
+	case TaskTypeExample:
+		return exampleStep2Str(s)
```

Comment:
Framework should not contains biz logic, can we register the step2Str function for each biz?

Reply:
it could be, but there's still biz `type` defined in framework. i don't want to register those constant related staff, we are just dig out of the design fault of storing them as int